Improved usability

RogerNogue opened the pull request:
Added some new buttons helpful for making objects static and navigable, fixed some bugs of the static objects and added some brofiler flags in order to help with profiling.
I would've splitted all these into different pull requests, but every PR is really hard to get accepted.
To test:
- load the graveyard scene
- select an object with childs that have mesh and  see the new buttons:
        - "make childs with mesh static"
        - "make childs with mesh navigable"
        - "make childs with mesh obstacles"
        - "add all navigable childs to the navmesh"
- select the game object "RecastMesh"
- click on "Make childs with mesh navigable"
- click on "Add all navigable childs to the navMesh"
- click on any of the navigable childs "floor" and in the "navigation" window scroll down and click on "generate navigability"

AlbertVVila commented on Source/GameObject.cpp:
```diff
@@ -289,6 +302,7 @@ void GameObject::DrawProperties()
 
 void GameObject::Update()
 {
+	PROFILE;
```
If you don't need it anymore it would be nice to remove it

AlbertVVila commented on Source/GameObject.cpp:
```diff
@@ -1282,7 +1296,7 @@ void GameObject::UpdateTransforms(math::float4x4 parentGlobal)
 bool GameObject::CheckDelete()
 {
-	//PROFILE;
+	PROFILE;
```
Same as before

AlbertVVila commented on Source/GameObject.cpp:
```diff
@@ -1305,3 +1319,89 @@ bool GameObject::CheckDelete()
+			{
+				child->makeObjectWithMeshStatic();
+				child->isStatic = true;
+				App->spacePartitioning->kDTree.Calculate();
```
Maybe we could simplify it only calling the *Calculate()* method once

AlbertVVila commented on Source/GameObject.cpp:
```diff
@@ -1305,3 +1319,89 @@ bool GameObject::CheckDelete()
+void GameObject::makeObjectWithMeshStatic()
+{
+	SetStaticAncestors();
+	App->scene->dynamicGOs.erase(this);
```
I think it would be more apropiate to use *DeleteFromSpacePartition* and *AddToSpacePartition*

AlbertVVila commented on Source/ModuleParticles.cpp:
```diff
@@ -139,6 +141,7 @@ bool ModuleParticles::Start()
 
 void ModuleParticles::Render(float dt, const ComponentCamera* camera) 
 {
+	PROFILE;
```
same as before

AlbertVVila commented on Source/ModuleScript.cpp:
```diff
@@ -47,6 +49,7 @@ bool ModuleScript::Init(JSON* config)
 
 update_status ModuleScript::Update(float dt)
 {
+	BROFILER_CATEGORY("Scripts Update", Profiler::Color::AliceBlue);
```
Same

RogerNogue commented:
> Not very aesthetic change. Occupies too much space and there is so much text. We should find a simpler way.

I fixed all the changes you requested and modified the texts of the buttons
![childs](https://user-images.githubusercontent.com/26113957/59681096-5dd6dc80-91d3-11e9-8a05-f7093e56609f.PNG)
